python/tskit/trees.py

```diff
@@ -2160,12 +2160,16 @@ def __build_newick(self, node, precision, node_labels):
             s = "("
             for child in self.children(node):
                 branch_length = self.branch_length(child)
-                subtree = self.__build_newick(child, precision, node_labels)
-                s += subtree + ":{0:.{1}f},".format(branch_length, precision)
+                subtree = self.__build_newick(
+                    child, precision, node_labels, branch_lengths
+                )
+                if branch_lengths:
+                    subtree += ":{0:.{1}f}".format(branch_length, precision)
+                s += subtree + ","
             s = s[:-1] + f"){label}"
         return s
 
-    def newick(self, precision=14, root=None, node_labels=None):
+    def newick(self, *, precision=14, root=None, node_labels=None, branch_lengths=True):
```

Comment:
This is a bit gratuitous I think and will probably break people's code (the newick method has been around a long time). I think making everything after ``precision`` keyword only is probably fine,though. (We can say in a comment that precision is positional for legacy only)